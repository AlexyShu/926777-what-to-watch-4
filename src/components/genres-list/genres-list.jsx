import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {ActionCreator} from "../../reducer.js";

const ALL_GENRES = `All genres`;

const GenresList = (props) => {
  const {films, onFilterClick} = props;
  const filters = [ALL_GENRES, ...new Set(films.map((film) => film.genre))];
  return (
    <ul className="catalog__genres-list">
      {filters.map((filter, i) => (
        <li onClick={() => onFilterClick(filter)} key={filter + i} className="catalog__genres-item catalog__genres-item--active">
          <a href="#" className="catalog__genres-link">
            {filter}
          </a>
        </li>)
      )}
    </ul>
  );
};


const mapDispatchToProps = (dispatch) => ({
  onFilterClick(genre) {
    dispatch(ActionCreator.changeGenre(genre));
  }
});

GenresList.propTypes = {
  films: PropTypes.arrayOf(
      PropTypes.shape({
        name: PropTypes.string.isRequired,
        posterSrc: PropTypes.string.isRequired,
        genre: PropTypes.string.isRequired,
        year: PropTypes.number.isRequired,
        ratingScore: PropTypes.string.isRequired,
        ratingLevel: PropTypes.string.isRequired,
        ratingCount: PropTypes.string.isRequired,
        descriptionPartOne: PropTypes.string.isRequired,
        descriptionPartTwo: PropTypes.string.isRequired,
        filmDirector: PropTypes.string.isRequired,
        filmStarring: PropTypes.string.isRequired,
      })
  ).isRequired,
  onFilterClick: PropTypes.func.isRequired,
};


export {GenresList};
export default connect(null, mapDispatchToProps)(GenresList);
