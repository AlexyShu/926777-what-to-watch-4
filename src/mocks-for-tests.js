export const mockFilmCard = {
  name: `The Grand Budapest`,
  genre: `Drama`,
  posterSrc: `img/the-grand-budapest-hotel-poster.jpg`,
  year: 2014,
  ratingScore: `8,9`,
  ratingLevel: `Very good`,
  ratingCount: `240 ratings`,
  descriptionPartOne: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
  presided over by concierge Gustave H. (Ralph Fiennes).
  Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
  descriptionPartTwo: `Gustave prides himself on providing first-class service to the hotel's guests,
  including satisfying the sexual needs of the many elderly women who stay there.
  When one of Gustave's lovers dies mysteriously,
  Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  filmDirector: `Director: Wes Andreson`,
  filmStarring: `Starring: Bill Murray, Edward Norton, Jude Law, Willem Dafoe and other`,
  video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  time: `1h 39m`,
  reviews: [
    {
      rating: 2,
      date: `June 29, 2020`,
      author: `Alexy Shu`,
      text: `This is an awesome movie! You should watch it!`
    },
    {
      rating: 3,
      date: `June 29, 2020`,
      author: `Alexy Shu`,
      text: `This is an awesome movie! You should watch it!`
    },
    {
      rating: 9,
      date: `June 29, 2020`,
      author: `Alexy Shu`,
      text: `This is an awesome movie! You should watch it!`
    }
  ]
};

export const mockFilms = [
  {
    id: 1,
    name: `The Grand Budapest Hotel`,
    posterUrl: `img/the-grand-budapest-hotel-poster.jpg`,
    previewUrl: `img/the-grand-budapest-hotel.jpg`,
    bigPosterUrl: `img/the-grand-budapest-hotel-bg.jpg`,
    backgroundColor: `#ffffff`,
    videoUrl: `https://some-link`,
    trailerUrl: `https://some-link`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    rating: 8.9,
    votes: 240,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: `99`,
    genre: `Comedy`,
    releaseYear: 2014,
    isFavorite: false
  },
  {
    id: 2,
    name: `The Grand Budapest Hotel`,
    posterUrl: `img/the-grand-budapest-hotel-poster.jpg`,
    previewUrl: `img/the-grand-budapest-hotel.jpg`,
    bigPosterUrl: `img/the-grand-budapest-hotel-bg.jpg`,
    backgroundColor: `#ffffff`,
    videoUrl: `https://some-link`,
    trailerUrl: `https://some-link`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    rating: 8.9,
    votes: 240,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: `99`,
    genre: `Drama`,
    releaseYear: 2014,
    isFavorite: false
  },
  {
    id: 3,
    name: `The Grand Budapest Hotel`,
    posterUrl: `img/the-grand-budapest-hotel-poster.jpg`,
    previewUrl: `img/the-grand-budapest-hotel.jpg`,
    bigPosterUrl: `img/the-grand-budapest-hotel-bg.jpg`,
    backgroundColor: `#ffffff`,
    videoUrl: `https://some-link`,
    trailerUrl: `https://some-link`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    rating: 8.9,
    votes: 240,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
    runTime: `99`,
    genre: `Action`,
    releaseYear: 2014,
    isFavorite: false
  }
];

export const mokcFunction = () => {};

export const MOCK_FILMS_COUNT = 8;

export const MOCK_ACTIVE_FILTER = `All genres`;

export const mockBool = false;
