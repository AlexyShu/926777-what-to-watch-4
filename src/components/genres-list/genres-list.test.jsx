import React from "react";
import renderer from "react-test-renderer";
import {GenresList} from "./genres-list.jsx";
import {mockFilms, mokcFunction, MOCK_ACTIVE_FILTER} from "../../mocks-for-tests.js";

describe(`Render correct GenresList`, () => {
  it(`Render GenresList`, () => {
    const tree = renderer
    .create(<GenresList
      films = {mockFilms}
      onFilterClick = {mokcFunction}
      activeFilter = {MOCK_ACTIVE_FILTER}
    />)
    .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
