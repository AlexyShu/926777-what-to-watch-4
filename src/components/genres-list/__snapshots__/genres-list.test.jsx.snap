// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correct GenresList Render GenresList 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item catalog__genres-item--active"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      All genres
    </a>
  </li>
  <li
    className="catalog__genres-item catalog__genres-item--active"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Fantastic
    </a>
  </li>
  <li
    className="catalog__genres-item catalog__genres-item--active"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Biography
    </a>
  </li>
  <li
    className="catalog__genres-item catalog__genres-item--active"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Drama
    </a>
  </li>
</ul>
`;
